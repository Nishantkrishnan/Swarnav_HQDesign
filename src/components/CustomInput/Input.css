.inputError {
  border-bottom: 1px solid #de1c1c !important;
}

.message {
  color: #de1c1c !important;
}
