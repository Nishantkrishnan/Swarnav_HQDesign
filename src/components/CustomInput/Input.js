import classNames from 'classnames';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './Input.css';

class Input extends Component {
 static propTypes = {
    className: PropTypes.string,
    type: PropTypes.string,
    placeHolder: PropTypes.string,
    onChange: PropTypes.func,
    onBlur: PropTypes.func,
    style: PropTypes.instanceOf(Object),
    message: PropTypes.string,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    error: PropTypes.bool,
    label: PropTypes.string,
    readonly: PropTypes.bool,
    disabled: PropTypes.bool,
  };
  static defaultProps = {
    type: '',
    style: {},
    className: '',
    value: undefined,
    error: false,
    disabled: false,
    readonly: false,
    placeHolder: '',
    message: '',
    label: '',
    onChange: f => f,
    onBlur: f => f,
  };

  constructor() {
    super();
  }
  render() {
    const {
      type,
      className,
      placeHolder,
      value,
      style,
      error,
      message,
      onChange,
      onBlur,
      label,
      ref,
      disabled,
      readonly,
    } = this.props;

    return (
      <div ref={ref} className={classNames(styles.inputWrapper)}>
        <label>{label}</label>
        <input
          type={type || 'text'}
          value={value}
          ref={ref => (this.referenceNode = ref)}
          className={classNames(className,styles.inputField, error && styles.inputError,{
            [styles.disabled]: disabled,
          })}
          onBlur={event => onBlur(event, this.referenceNode.value)}
          onChange={event => onChange(event, this.referenceNode.value)}
          placeholder={placeHolder}
          style={style}
          readOnly={readonly}
        />
        {error && message && <label className={classNames(styles.message)}>{message}</label>}
      </div>
    );
  }
}

export default Input;
