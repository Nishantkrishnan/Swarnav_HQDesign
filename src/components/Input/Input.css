.inputWrapper {
  position: relative;
  color: #000000;
}

.inputError {
  color: #de1c1c;
}

.disabled {
  pointer-events: none;
  opacity: 0.6;
}

.error {
  position: absolute;
  width: 6px;
  height: 100%;
  background: #de1c1c;
  top: 0;
  right: 0;
  border-top-right-radius: 5px;
  border-bottom-right-radius: 5px;
}

.message {
  display: block;
  position: absolute;
  left: 0;
  color: #de1c1c !important;
  top: 105%;
  font-size: 12px;
}

.messageLink {
  font-size: 14px;
  float: right;
  text-transform: capitalize;
  color: #5f5d5d;
}

.messageLink:hover {
  color: #000000;
}

.inputLabel {
  font-weight: 400;
  font-size: 12px;
  color: #50505a;
  letter-spacing: 0;
  margin-bottom: 10px;
  display: block;
  position: absolute;
  top: -25px;
}

.textarea {
  border: none;
  resize: none;
  outline: none;
}
