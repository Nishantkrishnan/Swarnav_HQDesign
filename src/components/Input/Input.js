import classNames from 'classnames';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './Input.css';

class Input extends Component {
  /*
  static propTypes = {
    className: PropTypes.string,
    type: PropTypes.string.isRequired,
    placeHolder: PropTypes.string,
    onChange: PropTypes.func,
    onEnter: PropTypes.func,
    onBlur: PropTypes.func,
    message: PropTypes.string,
    style: PropTypes.instanceOf(Object),
    maxLength: PropTypes.number,
    disabled: PropTypes.bool,
    onFocus: PropTypes.func,
    onClick: PropTypes.func,
    onKeyDown: PropTypes.func,
    defaultValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    error: PropTypes.bool,
    wrapperClass: PropTypes.string,
    label: PropTypes.string,
    readonly: PropTypes.bool,
  };
  static defaultProps = {
    type: '',
    style: {},
    className: '',
    defaultValue: undefined,
    value: undefined,
    error: false,
    maxLength: undefined,
    disabled: false,
    placeHolder: '',
    message: '',
    wrapperClass: '',
    label: '',
    readonly: false,
    onChange: f => f,
    onFocus: f => f,
    onBlur: f => f,
    onClick: f => f,
    onKeyDown: f => f,
    onEnter: f => f,
  };
  */
  constructor() {
    super();
    this.onKeyDown = this.onKeyDown.bind(this);
  }

  onKeyDown(event) {
    if (event.keyCode === 13 && this.props.onEnter) {
      event.preventDefault();
      this.props.onEnter(event.target.value);
    }
  }

  render() {
    const {
      type,
      className,
      placeHolder,
      defaultValue,
      value,
      error,
      message,
      maxLength,
      wrapperClass,
      ref,
      style,
      onBlur,
      onClick,
      onChange,
      label,
      disabled,
      readonly,
    } = this.props;

    return (
      <div ref={ref} className={classNames(styles.inputWrapper, wrapperClass)}>
        <label className={styles.inputLabel}>{label}</label>
        <input
          type={type || 'text'}
          ref={ref => (this.referenceNode = ref)}
          className={classNames(className,styles.inputField, error && styles.inputError, {
            [styles.disabled]: disabled,
          })}
          value={value}
          onKeyUp={this.onKeyDown}
          onBlur={(event) => { onBlur ? onBlur(event, this.referenceNode.value) : null }}
          onChange={event => onChange(event, this.referenceNode.value)}
          placeholder={placeHolder}
          defaultValue={defaultValue}
          maxLength={maxLength}
          style={style}
          readOnly={readonly}
          onClick={event => { onClick ? onClick(event, this.referenceNode.value, this.referenceNode.checked) : null }}
        />
        {error && <span className={classNames(styles.error)} />}

        {error && message && <label className={classNames(styles.message)}>{message}</label>}
      </div>
    );
  }
}

export default Input;
