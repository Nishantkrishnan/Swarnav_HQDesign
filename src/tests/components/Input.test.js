/**
 * Created by Sireesha on 15/11/17.
 */
// import 'jsdom-global/register';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../../components/Input/Input';

describe('Input component tests', () => {
  let InputTree;
  let wrapper;
  let disabledInput;
  const inputType = 'text';
  let changed = false;
  let enterkeyPressed = false;
  let escapeKeyPressed = false;
  let keyPressed = false;

  const eventFunction = (event) => {
    if (event !== undefined) changed = true;
  };

  const keyFunction = (event) => {
    if (event !== undefined && event.keyCode === 13) {
      enterkeyPressed = true;
    } else if (event !== undefined && event.keyCode === 27) {
      escapeKeyPressed = true;
    }
    keyPressed = true;
  };

  beforeEach(() => {
    changed = false;
    enterkeyPressed = false;
    escapeKeyPressed = false;
    keyPressed = false;
  });

  beforeAll(() => {
    InputTree = renderer.create(<Input
      type={inputType}
      placeHolder="Enter your name"
    />);

    wrapper = mount(<Input
      type={inputType}
      placeHolder="Enter your name"
      onKeyDown={keyFunction}
      onEnterKey={keyFunction}
      onEscapeKey={keyFunction}
      onClick={eventFunction}
      onChange={eventFunction}
      onFocus={eventFunction}
      onBlur={eventFunction}
    />);

    disabledInput = mount(<Input
      type={inputType}
      placeHolder="Enter your name"
      disabled
    />);
  });


  test('Snapshot match testing', () => {
    expect(InputTree).toMatchSnapshot();
  });

  test('Component structure testing', () => {
    expect(wrapper.find('input').node).toBeDefined();
  });

  test('Trigger onChange prop', () => {
    wrapper.find('input').simulate('change', { target: { validity: { valid: true } } });
    expect(changed).toBeTruthy();
  });

  test('Trigger onFocus prop', () => {
    wrapper.find('input').simulate('focus', { target: {} });
    expect(changed).toBeTruthy();
  });

  test('Trigger onBlur prop', () => {
    wrapper.find('input').simulate('blur');
    expect(changed).toBeTruthy();
  });

  test('Trigger onClick prop', () => {
    wrapper.find('input').simulate('click');
    expect(changed).toBeTruthy();
  });

  test('Trigger onEnterKey', () => {
    wrapper.find('input').simulate('keyDown', { keyCode: 13 });
    expect(enterkeyPressed).toBeTruthy();
  });

  test('Trigger onEscapeKey', () => {
    wrapper.find('input').simulate('keyDown', { keyCode: 27 });
    expect(escapeKeyPressed).toBeTruthy();
  });

  test('Trigger keyDown', () => {
    wrapper.find('input').simulate('keyDown', { keyCode: 15 });
    expect(keyPressed).toBeTruthy();
  });

  test('keyDown functional test for escape key', () => {
    wrapper.instance().keyDown({ keyCode: 27 });
    expect(escapeKeyPressed).toBeTruthy();
  });

  test('keyDown functional test for enter key', () => {
    wrapper.instance().keyDown({ keyCode: 13 });
    expect(enterkeyPressed).toBeTruthy();
  });

  test('disabledInput prop test', () => {
    expect(disabledInput.prop('disabled')).toBeTruthy();
  });

  test('disabledInput css class test', () => {
    expect(disabledInput.hasClass('disabled')).toBeTruthy();
  });

  test('should get value', () => {
    const value = wrapper.node.getValue();
    expect(wrapper.find('input').node.value).toMatch(value);
  });

  test('Key functional test', () => {
    expect(!wrapper.instance().getIsChecked()).toBeTruthy();
  });

  test('Default prop test onEscapeKey', () => {
    expect(Input.defaultProps.onEscapeKey('onEscapeKey')).toMatch('onEscapeKey');
  });

  test('Default prop test onKeyDown', () => {
    expect(Input.defaultProps.onKeyDown('onKeyDown')).toMatch('onKeyDown');
  });

  test('Default prop test onClick', () => {
    expect(Input.defaultProps.onClick('onClick')).toMatch('onClick');
  });

  test('Default prop test onEnterKey', () => {
    expect(Input.defaultProps.onEnter('onEnter')).toMatch('onEnter');
  });

  test('Default prop test onChange', () => {
    expect(Input.defaultProps.onChange('onChange')).toMatch('onChange');
  });

  test('Default prop test onFocus', () => {
    expect(Input.defaultProps.onFocus('onFocus')).toMatch('onFocus');
  });

  test('Default prop test onBlur', () => {
    expect(Input.defaultProps.onBlur('onBlur')).toMatch('onBlur');
  });

  test('Checked value test', () => {
    const checkBox = mount(<Input
      type="checkbox"
      checked
      placeHolder="Enter your name"
    />);

    expect(checkBox.instance().getIsChecked()).toBeTruthy();
  });

  test('defaultValue test', () => {
    const value = 'present';
    const wrap = mount(<Input
      type={inputType}
      placeHolder="Enter your name"
      defaultValue={value}
    />);
    expect(wrap.find('input').node.value).toMatch(value);
  });

  test('Value test', () => {
    expect(1 + 1).toBe(2);
    const value = 'present';
    const wrap1 = mount(<Input
      type={inputType}
      placeHolder="Enter your name"
      style={{ marginTop: '190px', marginLeft: '60px' }}
      value={value}
    />);
    expect(wrap1.find('input').node.value).toMatch(value);
  });
});
