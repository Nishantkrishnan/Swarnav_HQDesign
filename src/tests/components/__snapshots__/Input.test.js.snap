// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component tests Snapshot match testing 1`] = `undefined`;
